File: optimize-task/task.py
def find_sum_pairs_n2(nums, target):
    """Finds all pairs in a list that sum to a target, using nested loops (O(n^2) complexity)."""
    pairs = []
    for i in range(len(nums)):
        for j in range(i + 1, len(nums)):
            if nums[i] + nums[j] == target:
                pairs.append((nums[i], nums[j]))
    return pairs


# Question: Can you optimize this code to achieve a time complexity of O(N)?

File: optimize-task/task.js
/**
 * Finds a pair in given array that sums to a target
 * @param {Number[]} nums
 * @param {Number} target
 */
function findPairs(nums, target) {
  for (let i = 0; i < nums.length; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      if (nums[i] + nums[j] == target) {
        return [nums[i], nums[j]];
      }
    }
  }
}

// Question: Can you optimize this code to achieve a time complexity of O(N)?
